pkg/pool-cow/contracts/CowPoolFactory.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IPoolVersion } from "@balancer-labs/v3-interfaces/contracts/solidity-utils/helpers/IPoolVersion.sol";
+import { ICowPoolFactory } from "@balancer-labs/v3-interfaces/contracts/pool-cow/ICowPoolFactory.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import {
+    TokenConfig,
+    PoolRoleAccounts,
+    LiquidityManagement
+} from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { BasePoolFactory } from "@balancer-labs/v3-pool-utils/contracts/BasePoolFactory.sol";
+import { WeightedPool } from "@balancer-labs/v3-pool-weighted/contracts/WeightedPool.sol";
+import { Version } from "@balancer-labs/v3-solidity-utils/contracts/helpers/Version.sol";
+
+import { CowPool } from "./CowPool.sol";
+
+contract CowPoolFactory is ICowPoolFactory, IPoolVersion, BasePoolFactory, Version {
+    string internal _poolVersion;
+    address internal _trustedCowRouter;
+
+    constructor(
+        IVault vault,
+        uint32 pauseWindowDuration,
+        string memory factoryVersion,
+        string memory poolVersion,
+        address trustedCowRouter
+    ) BasePoolFactory(vault, pauseWindowDuration, type(CowPool).creationCode) Version(factoryVersion) {
+        _poolVersion = poolVersion;
+        _setTrustedCowRouter(trustedCowRouter);
+    }
+
+    /// @inheritdoc IPoolVersion
+    function getPoolVersion() external view override returns (string memory) {
+        return _poolVersion;
+    }
+
+    /// @inheritdoc ICowPoolFactory
+    function create(
+        string memory name,
+        string memory symbol,
+        TokenConfig[] memory tokens,
+        uint256[] memory normalizedWeights,
+        PoolRoleAccounts memory roleAccounts,
+        uint256 swapFeePercentage,
+        bytes32 salt
+    ) external override returns (address pool) {
+        if (roleAccounts.poolCreator != address(0)) {
+            revert StandardPoolWithCreator();
+        }
```

Comment:
Let's double check this with Daniel. It's probably OK though if COW only takes surplus fees.

Reply:
Daniel said that no pool creator is needed in the pool.